application/src/main/java/org/togetherjava/tjbot/commands/system/BotCore.java

```diff
@@ -94,54 +99,121 @@ public BotCore(JDA jda, Database database, Config config) {
-                    return uuid.toString();
-                })));
+        interactors.forEach(slashCommand -> slashCommand
+            .acceptComponentIdGenerator(((componentId, lifespan) -> {
+                UUID uuid = UUID.randomUUID();
+                componentIdStore.putOrThrow(uuid, componentId, lifespan);
+                return uuid.toString();
+            })));
 
         if (logger.isInfoEnabled()) {
-            logger.info("Available user interactors: {}", nameToInteractor.keySet());
+            logger.info("Available user interactors: {}", interactors);
         }
     }
 
+    /**
+     * Validates the given interactor
+     *
+     * @return whenever the command should be checked
+     */
+    @NotNull
+    private static Predicate<UserInteractor> validateInteractor() {
+        return interactor -> {
+            String name = interactor.getName();
+
+            if (name == null) {
+                logger.error("An interactor's name shouldn't be null! {}", interactor);
+                return false;
+            }
+
+            if (name.startsWith("s-") || name.startsWith("mc-") || name.startsWith("uc-")) {
+                logger.warn(
+                        "Names that starts with s-, mc- or uc- aren't recommended and might break other commands. (name: {})",
+                        name);
+            }
+
+            return true;
+        };
+    }
+
     @Override
-    public Collection<SlashCommand> getSlashCommands() {
-        return nameToInteractor.values()
-            .stream()
-            .filter(SlashCommand.class::isInstance)
-            .map(SlashCommand.class::cast)
-            .toList();
+    @Unmodifiable
+    public Collection<UserInteractor> getInteractors() {
+        return interactors;
     }
 
     @Override
-    public Optional<SlashCommand> getSlashCommand(String name) {
-        return Optional.ofNullable(nameToInteractor.get(name))
-            .filter(SlashCommand.class::isInstance)
-            .map(SlashCommand.class::cast);
+    public Optional<UserInteractor> getInteractor(final String name) {
+
+        return Optional.ofNullable(nameToInteractor.get(name));
+    }
+
+    public <T extends UserInteractor> Optional<T> getInteractor(final String name,
+            @Nullable final Class<? extends T> type) {
+        String prefix = "";
+
+        if (null == type) {
+            List<UserInteractor> anyInteractors = getAnyInteractors(name);
+            if (anyInteractors.isEmpty()) {
+                return Optional.empty();
+            } else {
+                return Optional.ofNullable((T) anyInteractors.get(0));
+            }
+        } else if (SlashCommand.class.isAssignableFrom(type)) {
+            prefix = "s-";
+        } else if (MessageContextCommand.class.isAssignableFrom(type)) {
+            prefix = "mc-";
+        } else if (UserContextCommand.class.isAssignableFrom(type)) {
+            prefix = "uc-";
+        }
```

Comment:
Duplication of the patterns. Would be resolved with the previously mentioned improvements on the prefix/id system